### ISSUE TITLE
signalR - openSignalingChannel not called

### ISSUE THREAD

**Issue by 9thScientist** [muaz-khan/RTCMultiConnection#444]
Hi,
My team and I are trying to build a video chat with multiple channels for 2 or more people. RTCMultiConnection will help us tremendously, but we are having problems when using SignalR as a private signalling server.

We were already using SignalR for this project so, when we decided to use RTCMultiConnection, we tried to extend our server so that it could also be the signalling server. We followed [your tutorial](https://github.com/muaz-khan/WebRTC-Experiment/blob/master/Signaling.md#how-to-use-signalr-for-signaling) and we used `https://our_ip:port/` as the `connection.socketURL`.

When we try to run the project and we get `socket.io is connected at:  "https://our_ip:port/"` on the browser's console. But the getUserMedia dialog doesn't show up and the call doesn't start. We tried using your severs as the `socketURL` and it worked perfectly. We are pretty sure the `openSignalingChannel` function is not being called with when using our signalling server.

Here is the client-side code:
```
var chat = $.connection.chatHub;
var rtcConnection = new RTCMultiConnection();
var onMessageCallBacks = {};

$.support.cors = true;

chat.client.onMessageReceived = function (message) {
    var message = JSON.parse(message);

    if (onMessageCallBacks[message.channel]) {
        onMessageCallBacks[message.channel](message.message);
    }
};

(...)

rtcConnection.openSignalingChannel = function (config) {
    var channel = config.channel || this.channel;

    onMessageCallBacks[channel] = config.onmessage;

    if (config.onopen) setTimeout(config.onopen, 1000);

    return {
        send: function (message) {
            message = JSON.stringify({
                message: message,
                channel: channel
            });

            // The server-side function is called broadcast
            chat.server.broadcast(message);
        }
    };
};
```

**Comment by muaz-khan** [comment]
Version 3 doesn't supports `openSignalingChannel`. Here is how to use SignalR with v3:

First step, modify [dev/SignalRConnection.js#L13](https://github.com/muaz-khan/RTCMultiConnection/blob/master/dev/SignalRConnection.js#L13) to set valid HUB urls:

```javascript
$.connection.hub.url = 'path-to-the-signalR-hub';
```

Now link the script in demo HTML files:

```html
<script src="/dev/SignalRConnection.js"></script>
```

Last step, use [`setCustomSocketHandler`](http://www.rtcmulticonnection.org/docs/setCustomSocketHandler/) to link the `SignalRConnection` object:

```javascript
connection.setCustomSocketHandler(SignalRConnection);

// rest of the code
connection.openOrJoin('roomid');
```

Now v3 will use `SignalRConnection` for signaling.

Doc: http://www.rtcmulticonnection.org/docs/setCustomSocketHandler/

Similar demo pages (that are using custom signaling:

* [demos/Firebase-Demo.html](https://github.com/muaz-khan/RTCMultiConnection/blob/master/demos/Firebase-Demo.html)
* [demos/PubNub-Demo.html](https://github.com/muaz-khan/RTCMultiConnection/blob/master/demos/PubNub-Demo.html)
* [demos/SSEConnection.html](https://github.com/muaz-khan/RTCMultiConnection/blob/master/demos/SSEConnection.html)

**Comment by 9thScientist** [comment]
Thank you,

We are still having problems, though. 

You start the hub in your [SignalRConnection.js](https://github.com/muaz-khan/RTCMultiConnection/blob/master/dev/SignalRConnection.js#L38), but since we are using SignalR as a big part of our project, we have the start of the hub before the `onBroadcastMessage`, so we deleted that line on your script. To ensure that `onBroadcastMessage` is available after the start of the hub, we defined it as  `hubProxy.on('broadcastMessage', function (chName, message) { ... })`.  This function is being called when every other person joins.

The only (big) problem is that, when another person joins, even though it calls the `broadcastMessage` function, it won't call `onStream` function, so we don't get the mediaElement from the other user.

Any idea of what might be causing it?
